### PR TITLE
Added the rest of the STIGs

### PR DESCRIPTION
**Pull request by n-boshnakov**
**What this PR does / why we need it**:
This PR:
- Adds the rest of the STIGs to the Kubernetes Hardening guide
- Renames the "Compliance" section to "Security and Compliance"
- Adds an alias

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
The newly added section needs to be updated with additional information - Secure By Default and Comments.
@dimityrmirchev's help would be appreciated here.

**Release note**:

```improvement developer
Added STIGs for Service Providers in the Kubernetes Hardening guide
```

### CODE REVIEW

**Comment by JordanJordanov** [comment]
@dimityrmirchev Shall we keep the "Secure By Default" and "Comments" columns for the DISA STIGs rules for `Service Provider`?

File: website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+|242425|Kubernetes Kubelet must enable tlsCertFile for client authentication to secure service.|||
+|242426|Kubernetes etcd must enable client authentication to secure service.|||
+|242427|Kubernetes etcd must have a key file for secure communication.|||
+|242428|Kubernetes etcd must have a certificate for communication.|||
+|242429|Kubernetes etcd must have the SSL Certificate Authority set.|||
+|242430|Kubernetes etcd must have a certificate for communication.|||
+|242431|Kubernetes etcd must have a key file for secure communication.|||
+|242432|Kubernetes etcd must have peer-cert-file set for secure communication.|||
+|242433|Kubernetes etcd must have a peer-key-file set for secure communication.|||
+|242434|Kubernetes Kubelet must enable kernel protection.|||
```
Comment (dimityrmirchev) [inline website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md:94]: This rule is already mentioned in the "Control Plane Configuration" section so we can remove it from here.

```suggestion
```

File: website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+tags: ["task"]
+aliases: ["/docs/guides/kubernetes-hardening/"]
+---
+
+## Overview
+The Gardener team takes security seriously, which is why we mandate the Security Technical Implementation Guide (STIG) for Kubernetes as published by the Defense Information Systems Agency (DISA) [here](https://public.cyber.mil/stigs/downloads/). We offer Gardener adopters the opportunity to show compliance with DISA Kubernetes STIG via the compliance checker tool [diki](https://github.com/gardener/diki). The latest release in machine readable format can be found in the [STIGs Document Library](https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=container-platform) by searching for Kubernetes.
+
+## Kubernetes Clusters Security Requirements
+
+[DISA Kubernetes STIG version 1 release 10](https://cyber.trackr.live/stig/Kubernetes/1/10) contains 93 rules overall. **Only the following rules, however, apply to you**. Some of them are secure-by-default, so your responsibility is to make sure that they are not changed. For your convenience, the requirements are grouped logically and per role:
```
Comment (dimityrmirchev) [inline website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md:16]: Let's refer to the latest revision.

```suggestion
[DISA Kubernetes STIG version 1 release 11](https://cyber.trackr.live/stig/Kubernetes/1/11) contains 91 rules overall. **Only the following rules, however, apply to you**. Some of them are secure-by-default, so your responsibility is to make sure that they are not changed. For your convenience, the requirements are grouped logically and per role:
```

**Review by dimityrmirchev — COMMENTED** [review]
Some minor remarks from my side. Other looks good.

> Shall we keep the "Secure By Default" and "Comments" columns for the DISA STIGs rules for Service Provider?

I think that we can remove these columns for now. If there is the need we can always add them later on.

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Review by JordanJordanov — APPROVED** [review]
/lgtm

**Pull request merged**